### PR TITLE
[model] feat: Add qwen3_5_moe support (language model only)

### PR DESCRIPTION
**Pull request by piyifan123**
##  Summary
                                                                                                                                                                                      
  - Add Qwen3.5-MoE support — new patched modeling for qwen3_5_moe with fused MoE dispatch (fc1_1_2_weight merged gate_up_proj), varlen flash linear attention, Ulysses sequence
  parallelism, fused cross-entropy loss, and expert parallel plan.                                                                                                                    
  - Deduplicate patchgen configs via name_map — extend the patchgen framework (PatchConfig, codegen) with a name_map text-substitution mechanism so that qwen3_5_moe reuses
  GatedDeltaNet __init__, _get_local_conv1d_weight, and forward patches from qwen3_5 with {"Qwen3_5": "Qwen3_5Moe"}, eliminating ~268 lines of duplication.                           
  - Fix fused MoE autograd view error — clone the output of reshape() in TritonFusedMoeExpertFunction and MergedFc1TritonFusedMoeExpertFunction to avoid PyTorch's "in-place op on
  view from custom Function" error when downstream ops (e.g. SparseMoeBlock's +=) modify the tensor in-place.                                                                         
  - Fix e2e test weight init for Qwen3.5-MoE — HF transformers v5's save_pretrained() splits stacked 3D expert tensors into per-expert keys via revert_weight_conversion(), which
  VeOmni's load_model_weights can't merge back. Use save_original_format=False for qwen3_5_moe to save in native stacked format.                                                      
  - Clean up qwen3_5 patches — remove unused slice_position_embedding patch and TextModel.forward override (no longer needed after Ulysses SP).
  
##  TODO in follow up PRs

1. @Luosuu Support EP for fused gate_up weights in MoE op.
2. Lu Di: Support ViT.

                                                                                                                                                                                      
##  New files                 
                                                                                                                                                                                      
  - veomni/models/transformers/qwen3_5_moe/ — __init__.py, parallel_plan.py, qwen3_5_moe_gpu_patch_gen_config.py, generated/                                                          
  - tests/toy_config/qwen3_5_moe_toy/ — toy config for e2e testing
  - docs/examples/qwen3_5.md — example training command                                                                                                                               
                                                                                                                                                                                      
 ## Test plan                                                                                                                                                                           
                                                                                                                                                                                      
  - pytest -sx tests/models/test_models_patch.py -k "qwen3_5" — verify generated code matches for both qwen3_5 and qwen3_5_moe                                                        
  - pytest -sx tests/e2e/test_e2e_parallel.py -k "qwen3_5_moe" — e2e SP/EP consistency (requires GPU)
  - Regenerate and verify no diff: python -m veomni.patchgen.run_codegen veomni.models.transformers.qwen3_5_moe.qwen3_5_moe_gpu_patch_gen_config -o                                   
  veomni/models/transformers/qwen3_5_moe/generated --diff                                                                                                                             

## Sample training run

8XB200

<img width="954" height="742" alt="image" src="https://github.com/user-attachments/assets/196b72d6-01ec-4ff6-b201-d6a05f00e5ab" />

### CODE REVIEW

File: veomni/ops/fused_moe/group_gemm.py [hunk 1]
```diff
@@ -103,7 +103,11 @@ def forward(
         expert_output = moe_gather(fc2_output, scatter_index)
 
         # reshape the output with input shape
-        output = expert_output.reshape(hidden_states.shape)
+        # Clone to avoid returning a view from a custom autograd Function.
+        # Downstream ops (e.g. SparseMoeBlock's `+=`) modify this in-place,
+        # which PyTorch forbids on views created by custom Functions.
+        output = expert_output.reshape(hidden_states.shape).clone()
```
Comment (piyifan123) [inline veomni/ops/fused_moe/group_gemm.py]: @Luosuu please check this.

Comment (Luosuu) [reply veomni/ops/fused_moe/group_gemm.py]: I do not get the point.. can you explain more?

Comment (Luosuu) [reply veomni/ops/fused_moe/group_gemm.py]: let's do
```python
hidden_states = hidden_states + self.shared_experts(residuals)
```
instead of 
```python
expert_output += shared_expert_output
```
in Qwen3_5 SparseMoeBlock.forward

**Review by Luosuu — DISMISSED** [review]

File: veomni/models/auto.py [hunk 2]
```diff
@@ -98,6 +98,10 @@ def build_foundation_model(
         if moe_implementation not in ["eager", "fused"]:
             raise ValueError(f"Invalid moe_implementation: {moe_implementation}")
         config._moe_implementation = moe_implementation
+        # Propagate to text_config for VLM-style models (e.g. qwen3_5_moe)
+        # so that sub-modules like Experts can read _moe_implementation.
+        if hasattr(config, "text_config"):
```
Comment (Coach257) [inline veomni/models/auto.py]: This may cause error when other vlm except qwenvl init text_model directly from config. Can we patch this in QwenMoeModel.__init__?
https://github.com/huggingface/transformers/blob/aad13b87ed59f2afcfaebc985f403301887a35fc/src/transformers/models/qwen3_vl_moe/modeling_qwen3_vl_moe.py#L1098
https://github.com/huggingface/transformers/blob/aad13b87ed59f2afcfaebc985f403301887a35fc/src/transformers/models/qwen3_5_moe/modeling_qwen3_5_moe.py#L1540

Comment (piyifan123) [reply veomni/models/auto.py]: done

File: veomni/ops/fused_moe/group_gemm.py (hunk 1 above)
Comment (piyifan123) [reply veomni/ops/fused_moe/group_gemm.py]: good idea. done

**Comment by piyifan123** [comment]
CI machine is in high demand now. merging now since we have verified locally for key tests affected by this change.
image

File: veomni/models/transformers/qwen3_5_moe/qwen3_5_moe_gpu_patch_gen_config.py [hunk 3]
```diff
@@ -0,0 +1,411 @@
...
+        from fla.modules.convolution import causal_conv1d as causal_conv1d_fn
+        from fla.modules.convolution import causal_conv1d_update
+        from fla.ops.gated_delta_rule import chunk_gated_delta_rule, fused_recurrent_gated_delta_rule
+    except ImportError:
+        chunk_gated_delta_rule, fused_recurrent_gated_delta_rule = None, None
+        FusedRMSNormGated = None
+        causal_conv1d_update, causal_conv1d_fn = None, None
+        logging.get_logger(__name__).warning(
+            "Failed to import FLA modules: fallback to eager implementation. "
+            "This case can't support rmpad_with_pos_ids=True!"
```
Comment (Coach257) [inline veomni/models/transformers/qwen3_5_moe/qwen3_5_moe_gpu_patch_gen_config.py:90]: Why is rmpad not supported. VeOmni now only has `rmpad_with_pos_ids` mode. @piyifan123

Comment (piyifan123) [reply veomni/models/transformers/qwen3_5_moe/qwen3_5_moe_gpu_patch_gen_config.py:90]: The line triggers when fla is not in the env. Only fla supports varlen gdn (i.e., rmpad with pos id case), and torch eager impl does not support varlen
